src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -2419,6 +2419,25 @@ protected AssemblySymbol GetForwardedToAssembly(string fullName, BindingDiagnost
             return null;
         }
 
+        internal static ContextualAttributeBinder TryGetContextualAttributeBinder(Binder binder)
+        {
+            if ((binder.Flags & BinderFlags.InContextualAttributeBinder) != 0)
+            {
+                do
+                {
+                    if (binder is ContextualAttributeBinder contextualAttributeBinder)
+                    {
+                        return contextualAttributeBinder;
+                    }
+
+                    binder = binder.Next;
+                }
+                while (binder != null);
```

Comment:
>while (binder != null); [](http://example.com/codeflow?start=16&length=23)

Would it be worth adding an assert after the loop? It looks like we expect to never reach that point.